firefox-ios/Client/Frontend/Browser/TermsOfUse/TermsOfUseCoordinator.swift

```diff
@@ -88,8 +89,8 @@ final class TermsOfUseCoordinator: BaseCoordinator, TermsOfUseCoordinatorDelegat
         guard didShowThisLaunch else { return true }
 
         // 4. If shown this launch, show it if enough time has passed since dismissal.
-        guard let dismissedWithoutAcceptDate = defaults.object(forKey:
-                TermsOfUseMiddleware.DefaultKeys.dismissedWithoutAcceptDate) as? Date else { return false }
+        guard let dismissedWithoutAcceptDate: Date = prefs.objectForKey(PrefsKeys.TermsOfUseDismissedDate)
```

Comment:
I think we should save a timestamp and use `timestampForKey` here.

Reply:
yes, I added this change